src/routing/routing_heuristics.jl

```diff
@@ -292,50 +292,59 @@ function optimize_waypoints(
     # Initialize variables for objective function
     wpts_length::Int64 = length(waypoints_initial) + 2 # +2 for depot start/end points
     wpts::Vector{Point{2,Float64}} = Vector{Point{2,Float64}}(undef, wpts_length)
+    wpts[1] = last_ms_route.route.nodes[1] # Depot start fixed
+    wpts[end] = last_ms_route.route.nodes[end] # Depot end fixed
+
     has_bad_waypoint::BitVector = falses(wpts_length)
     exclusion_count::Int64 = 0
     score::Float64 = 0.0
 
-    best_soln = Ref{MSTSolution}(soln)
-    best_score = Ref(Inf)
-    best_count = Ref(0)
+    best_soln::MSTSolution = soln
+    best_score::Float64 = critical_path(soln, vessel_weightings)
+    best_count::Int64 = 0
+    soln_proposed::MSTSolution = soln
```

Comment:
Commit message speaks to "performance" but the removal of `Ref` shouldn't have had any effect really except maybe make the code a tiny bit slower.

Adding `soln_proposed` here doesn't "preallocate memory" any more than previously, though I guess you could argue listing all the variables that are defined out of scope from `obj()` improves clarity.

Okay to leave as is but flagging that I'm not unsure of the intent here.